### ISSUE TITLE
Question regarding Batchnorm

### ISSUE THREAD

**Issue by hw-kim-ivan** [aserdega/convlstmgru#5]
Thank you for offering great code:)

I have 2 questions regarding batchnormalization in your code.
As shown in convlstm.py's batchnorm part like below

  def reset_parameters(self):
      #self.conv.reset_parameters()
      nn.init.xavier_uniform_(self.conv.weight, gain=nn.init.calculate_gain('tanh'))
      self.conv.bias.data.zero_()
  
      if self.batchnorm:
          self.bn1.reset_parameters()
          self.bn2.reset_parameters()

Q1) Does it mean that batchnorm is operating between (vertically) convlstm layer? 
or between (horizontally) convlstm cell?

Q2) Is it really work as batchnormalization? As I know of, xavier uniform is not exactly same as batchnormalization.

My question might be wrong, thus waiting for your merciful answer:)

**Comment by aserdega** [comment]
Hello. 

So, first thing is that **nn.init.xavier_uniform_(self.conv.weight, gain=nn.init.calculate_gain('tanh'))** does initialization for parameters of convolutional module, not for batch norm. 
Just take a more careful look at the parameters that are passed into initializer :)

And regarding batch norm, it has been a while since I've wrote this code. I don't see that bn modules are really used inside of RNN. I probably just forgot to delete this modules from cell class...

But if you gonna consider using batch norm, you want to apply it to the final output of the cell. In the very end of _forward_ function.

Feel free to ask if there is anything else not clear to you. And if you find my code useful, I'd appreciate if you star the repo :))
